kedro/extras/datasets/pandas/gbq_dataset.py

```diff
@@ -161,3 +169,145 @@ def _validate_location(self):
+    BigQuery. It uses ``pandas.read_gbq`` which itself uses ``pandas-gbq``
+    internally to read from BigQuery table. Therefore it supports all allowed
+    pandas options on ``read_gbq``.
+
+    Example adding a catalog entry with
+    `YAML API <https://kedro.readthedocs.io/en/stable/05_data/\
+        01_data_catalog.html#using-the-data-catalog-with-the-yaml-api>`_:
+
+    .. code-block:: yaml
+
+        >>> vehicles:
+        >>>   type: pandas.GBQQueryDataSet
+        >>>   sql: "select shuttle, shuttle_id from spaceflights.shuttles;"
+        >>>   project: my-project
+        >>>   credentials: gbq-creds
+        >>>   load_args:
+        >>>     reauth: True
+        >>>   save_args:
+        >>>     chunk_size: 100
+
+
+    Example using Python API:
+    ::
+
+        >>> from kedro.extras.datasets.pandas import GBQQueryDataSet
+        >>>
+        >>> sql = "SELECT * FROM dataset_1.table_a"
+        >>>
+        >>> data_set = GBQQueryDataSet(sql, project='my-project')
+        >>>
+        >>> sql_data = data_set.load()
+        >>>
+    """
+
+    DEFAULT_LOAD_ARGS = {}  # type: Dict[str, Any]
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        sql: str = None,
+        project: str = None,
+        credentials: Union[Dict[str, Any], Credentials] = None,
+        load_args: Dict[str, Any] = None,
+        fs_args: Dict[str, Any] = None,
+        filepath: str = None,
+    ) -> None:
+        """Creates a new instance of ``GBQQueryDataSet``.
+
+        Args:
+            sql: The sql query statement.
+            project: Google BigQuery Account project ID.
+                Optional when available from the environment.
+                https://cloud.google.com/resource-manager/docs/creating-managing-projects
+            credentials: Credentials for accessing Google APIs.
+                Either ``google.auth.credentials.Credentials`` object or dictionary with
+                parameters required to instantiate ``google.oauth2.credentials.Credentials``.
+                Here you can find all the arguments:
+                https://google-auth.readthedocs.io/en/latest/reference/google.oauth2.credentials.html
+            load_args: Pandas options for loading BigQuery table into DataFrame.
+                Here you can find all available arguments:
+                https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_gbq.html
+                All defaults are preserved.
+            fs_args: Extra arguments to pass into underlying filesystem class constructor
+                (e.g. `{"project": "my-project"}` for ``GCSFileSystem``) used for reading the
+                SQL query from filepath.
+            filepath: A path to a file with a sql query statement.
+
+        Raises:
+            DataSetError: When ``sql`` and ``filepath`` parameters are either both empty or both provided, as well as when the `save()` method is invoked.
```

Comment:
```suggestion
            DataSetError: When ``sql`` and ``filepath`` parameters are either both empty or both provided, 
            as well as when the `save()` method is invoked.
```

It's a really long line, so breaking it two makes it more readable.